### ISSUE TITLE
Incorrect Symbol for ResourceMethod from bala project

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#36891]
**Description:**
Consider the following scenario where I'm trying to retrieve the symbol of the only `client resource method` depicted below.

```Ballerina
public client class Client {
    public string url;

    public function init(string url) {
        self.url = url;
    }

    remote function post(@untainted string path, RequestMessage message, string targetType="xml") returns Response {
        Response response = new();
        return response;
    }
    
    resource function post path1/[string id1]/path2/[string... ids](string str, string... ids2) returns Response {
        return new Response();
    }
}

public class Response {
    public int statusCode = 200;
    public string reasonPhrase = "Test Reason phrase";
}

public type RequestMessage string|xml|json|byte[]|();
```

The following captures depicts the difference of the symbol when the above client is obtained from a `source project` vs `bala projet`. 

From source project
![image](https://user-images.githubusercontent.com/35211477/177617696-e2380179-ac7b-417b-9725-7b80a99c1048.png)

From bala project
![image](https://user-images.githubusercontent.com/35211477/177618025-0b6c2295-a894-45bc-b392-d134f2fce803.png)

Seems that there is something wrong with the  BIR for ResourceMethodSymbol here.

**Steps to reproduce:**

**Affected Versions:**
Mentioned above

**Comment by dulajdilshan** [comment]
Yeah, The BIR changes needed to be done

**Comment by mohanvive** [comment]
@dulajdilshan Haven't BIR changes are completed by Compiler FE team?

**Comment by dulajdilshan** [comment]
> @dulajdilshan Haven't BIR changes are completed by Compiler FE team?

No, they haven't it seems. @rdulmina Can you please share any updates on this?

**Comment by rdulmina** [comment]
Fixed with  #36959
